src/ptvsd/adapter/server.py

```diff
@@ -45,12 +161,32 @@ class Capabilities(components.Capabilities):
             "supportedChecksumAlgorithms": [],
         }
 
-    def __init__(self, session, stream):
-        super(Server, self).__init__(session, stream)
+    def __init__(self, session, connection):
+        with _lock:
+            connection.server = self
+        self.connection = connection
 
-        self.pid = None
+        super(Server, self).__init__(session, channel=connection.channel)
+
+        self.pid = connection.pid
```

Comment:
Maybe the `pid` and `ppid` could be properties that always got the value from the connection? (it doesn't seem like the connection is ever set to None afterwards)